source/event_loop.c

```diff
@@ -94,6 +95,51 @@ void aws_event_loop_group_clean_up(struct aws_event_loop_group *el_group) {
     aws_array_list_clean_up(&el_group->event_loops);
 }
 
+struct aws_event_loop_group_destroy_async_data {
+    struct aws_allocator *allocator;
+    struct aws_event_loop_group *el_group;
+    aws_event_loop_group_cleanup_complete_fn *completion_callback;
+    void *user_data;
+};
+
+static void s_event_loop_destroy_async_thread_fn(void *thread_data) {
+    struct aws_event_loop_group_destroy_async_data *completion_data = thread_data;
+
+    aws_event_loop_group_clean_up(completion_data->el_group);
+
+    completion_data->completion_callback(completion_data->user_data);
+
+    aws_mem_release(completion_data->allocator, thread_data);
+}
+
+void aws_event_loop_group_cleanup_async(
+    struct aws_event_loop_group *el_group,
+    aws_event_loop_group_cleanup_complete_fn completion_callback,
+    void *user_data) {
+    struct aws_thread cleanup_thread;
+    AWS_ZERO_STRUCT(cleanup_thread);
+
+    struct aws_event_loop_group_destroy_async_data *data =
+        aws_mem_calloc(el_group->allocator, 1, sizeof(struct aws_event_loop_group_destroy_async_data));
+    AWS_FATAL_ASSERT(data != NULL);
+
+    data->allocator = el_group->allocator;
+    data->el_group = el_group;
+    data->completion_callback = completion_callback;
+    data->user_data = user_data;
+
+    AWS_FATAL_ASSERT(aws_thread_init(&cleanup_thread, el_group->allocator) == AWS_OP_SUCCESS);
+
+    struct aws_thread_options thread_options;
+    AWS_ZERO_STRUCT(thread_options);
+
+    AWS_FATAL_ASSERT(
+        aws_thread_launch(&cleanup_thread, s_event_loop_destroy_async_thread_fn, data, &thread_options) ==
+        AWS_OP_SUCCESS);
+
+    aws_thread_clean_up(&cleanup_thread);
```

Comment:
We don’t need to join?

Reply:
Nevermind, I’m dumb, you intentionally detached so it would stay async

Reply:
clean_up = detach